### ISSUE TITLE
Last character is sometimes dropped when using Variable Fonts

### ISSUE THREAD

**Issue by justvanrossum** [typemytype/drawbot#315]
Depending on the axis settings, DrawBot sometimes drops the last letter from a string as set with `text()`.

@LettError 
Using `MutatorSans.ttf` font from https://github.com/LettError/mutatorSans

```python
txt = '→→→→'

values = [('ok', 500), ('dropping', 502)]
weightValue = 200

for tag, widthValue in values:    
    newPage(690, 240)
    
    font("MutatorMathTest")
    fontSize(200)
    fontVariations(wght=weightValue, wdth=widthValue)
    w, h = textSize(txt)
    fill(0,0.8,0.8,0.3)
    rect(20, 20, w, h)

    fill(0.8,0,0.8,0.3)
    text(txt, (20, 20), align="left")
    saveImage('drawbot_drops_letter_%s.jpg' % tag)
```

![drawbot_drops_letter_dropping](https://user-images.githubusercontent.com/4246121/67683122-0f132d80-f991-11e9-8a7c-06535e05f239.jpg)
![drawbot_drops_letter_ok](https://user-images.githubusercontent.com/4246121/67683134-15090e80-f991-11e9-968b-0d8b4eb8dcd9.jpg)

**Comment by LettError** [comment]
Thank you for posting this, @justvanrossum GitHub had the hiccups when I tried earlier.

**Comment by typemytype** [comment]
What happens inside: `text(..)` draws the text actually in a `textBox(..)` with a width/height its gets from `textSize`.

Somehow the frame setter thinks the text does not fit inside that box.... and drops the last character.

I assume there is a text wrap setting that can be set while drawBot is setting text with `text(..)`

**Comment by justvanrossum** [comment]
Erik, can you verify whether BezierPath().text() shows the same problem?

**Comment by LettError** [comment]
I was going to and then got lost trying to set feature variation values for BezierPath.

**Comment by LettError** [comment]
Same result when drawing text with bezierPath().text().
 * Blue background: drawing text directly.
 * Green background: drawing with bezierPath().

```python

weightMin = listFontVariations('MutatorMathTest')['wght']['minValue']
weightMax = listFontVariations('MutatorMathTest')['wght']['maxValue']

widthMin = listFontVariations('MutatorMathTest')['wdth']['minValue']
widthMax = listFontVariations('MutatorMathTest')['wdth']['maxValue']

txt = '→→→→'

thisFontName = "MutatorMathTest"
thisSize = 200
thisBackgroundFill = (0,0.8,0.8,0.3)
thisBackgroundFill2 = (0.3,0.8,0,0.3)
thisFill = (0.8,0,0.8,0.3)

values = [('dropping', 500), ('ok', 502)]
weightValue = 200

for tag, widthValue in values:
    # direct
    newPage(690, 240)    
    font(thisFontName)
    fontSize(thisSize)
    fontVariations(wght=weightValue, wdth=widthValue)
    w, h = textSize(txt)
    fill(*thisBackgroundFill)
    rect(20, 20, w, h)

    fill(*thisFill)
    text(txt, (20, 20), align="left")
    saveImage('drawbot_drops_letter_%s.jpg' % tag)

    # bezierPath
    newPage(690, 240)
    fill(*thisBackgroundFill2)
    rect(20, 20, w, h)
    s = FormattedString()
    s.append(
        txt, 
        font=thisFontName,
        fontSize=thisSize,
        fontVariations=dict(wght=weightValue, wdth=widthValue),
        fill=0
        )
    fill(*thisFill)
    B = BezierPath()    
    B.text(s, (20, 20))
    drawPath(B)
    saveImage('drawbot_drops_letter_%s_bezierPath.jpg' % tag)
```

![drawbot_drops_letter_ok_bezierPath](https://user-images.githubusercontent.com/1243961/67697729-4db4e200-f9a9-11e9-942b-0f4ca58db3af.jpg)
![drawbot_drops_letter_ok](https://user-images.githubusercontent.com/1243961/67697730-4db4e200-f9a9-11e9-88df-cadac3158ede.jpg)
![drawbot_drops_letter_dropping_bezierPath](https://user-images.githubusercontent.com/1243961/67697731-4e4d7880-f9a9-11e9-8bd8-765bb82764c1.jpg)
![drawbot_drops_letter_dropping](https://user-images.githubusercontent.com/1243961/67697732-4e4d7880-f9a9-11e9-875b-e19fcd852b7c.jpg)

**Comment by LettError** [comment]
Also happens when aligning to the right.

**Comment by justvanrossum** [comment]
This may be related to #254 which has been open for a long time.

**Comment by justvanrossum** [comment]
Should be fixed with #319
